### PR TITLE
Use function 'ConvertResourceGroupNameToLower' from 'cloud-provider-azure' provider pkg

### PR DESCRIPTION
**Pull request by chiukapoor**
Signed-off-by: Chirayu Kapoor <dev.csociety@gmail.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Updates the code to use function 'ConvertResourceGroupNameToLower' from 'cloud-provider-azure' provider pkg

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2892

**Special notes for your reviewer**:

None

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by primeroz** [comment]
I have 2 prs using the old version ( they are alternative to each other so only one will be merged in case ) 

* https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2894
* https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2821

should i just change to the same here ?

**Comment by CecileRobertMichon** [comment]
@primeroz we'll merge your bug fix first so it can be cherry-picked cleanly, then rebase this and merge it.

Thanks @chiukapoor for opening the PR. LGTM, pending merging the bug fix mentioned in the PR above.

/hold

**Comment by chiukapoor** [comment]
@CecileRobertMichon while adding the `sigs.k8s.io/cloud-provider-azure` dependency couple of existing dependencies are getting updated to newer versions, which in turn is causing the tests to fail.

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/54b3635996f4ba22b31833eb8f3a6e9256daa85f/go.mod

```
test/e2e/aks.go:222:5: too many arguments in call to Eventually
        have (func() bool, []interface{})
        want (...interface{}) (typecheck)
        }, intervals...).Should(Equal(true), "System machine pools not detected")
           ^
test/e2e/aks.go:529:5: too many arguments in call to Eventually
        have (func(g "github.com/onsi/gomega/types".Gomega), []interface{})
        want (...interface{}) (typecheck)
        }, inputGetter().WaitIntervals...).Should(Succeed())
           ^
test/e2e/aks.go:538:5: too many arguments in call to Eventually
        have (func(g "github.com/onsi/gomega/types".Gomega), []interface{})
        want (...interface{}) (typecheck)
        }, inputGetter().WaitIntervals...).Should(Succeed())
           ^
test/e2e/aks.go:583:5: too many arguments in call to Eventually
        have (func() error, []interface{})
        want (...interface{}) (typecheck)
        }, input.WaitIntervals...).Should(Succeed(), "failed to create public IP prefix")
           ^
test/e2e/aks.go:636:6: too many arguments in call to Eventually
        have (func() bool, []interface{})
        want (...interface{}) (typecheck)
                }, input.WaitIntervals...).Should(BeTrue(), "Deleted MachinePool %s/%s still exists", machinePool.Namespace, machinePool.Name)
                   ^
test/e2e/aks.go:641:6: too many arguments in call to Eventually
        have (func() bool, []interface{})
        want (...interface{}) (typecheck)
                }, input.WaitIntervals...).Should(BeTrue(), "Deleted AzureManagedMachinePool %s/%s still exists", infraMachinePool.Namespace, infraMachinePool.Name)
                   ^
test/e2e/aks.go:654:5: too many arguments in call to Eventually
        have (func(g "github.com/onsi/gomega/types".Gomega), []interface{})
        want (...interface{}) (typecheck)
        }, input.WaitIntervals...).Should(Succeed())
           ^
test/e2e/aks.go:669:5: too many arguments in call to Eventually
        have (func(g "github.com/onsi/gomega/types".Gomega), []interface{})
        want (...interface{}) (typecheck)
        }, input.WaitIntervals...).Should(Succeed())
           ^
test/e2e/azure_clusterproxy.go:108:18: acp.GetWorkloadCluster undefined (type *AzureClusterProxy has no field or method GetWorkloadCluster) (typecheck)
        workload := acp.GetWorkloadCluster(ctx, namespace, name)
                        ^
test/e2e/azure_clusterproxy.go:221:23: acp.GetClient undefined (type *AzureClusterProxy has no field or method GetClient) (typecheck)
        clusterClient := acp.GetClient()
                             ^
test/e2e/helpers.go:119:5: too many arguments in call to Eventually
        have (func() bool, []interface{})
        want (...interface{}) (typecheck)
        }, intervals...).Should(BeTrue(), func() string { return DescribeFailedDeployment(ctx, input) })
           ^
test/e2e/helpers.go:187:5: too many arguments in call to Eventually
        have (func() bool, []interface{})
        want (...interface{}) (typecheck)
        }, intervals...).Should(BeTrue(), func() string { return DescribeFailedJob(ctx, input) })
           ^
test/e2e/helpers.go:282:5: too many arguments in call to Eventually
        have (func() bool, []interface{})
        want (...interface{}) (typecheck)
        }, intervals...).Should(BeTrue(), func() string { return DescribeFailedDaemonset(ctx, input) })
           ^
test/e2e/helpers.go:340:5: too many arguments in call to Eventually
        have (func() bool, []interface{})
        want (...interface{}) (typecheck)
        }, intervals...).Should(BeTrue(), func() string { return DescribeFailedService(ctx, input) })
```

**Comment by CecileRobertMichon** [comment]
> while adding the sigs.k8s.io/cloud-provider-azure dependency, a couple of existing dependencies are getting updated to newer versions, which in turn is causing the tests to fail.

Some of these dependencies are getting updated in #2773, let's see if we can rebase this on top of it once it merges

**Comment by CecileRobertMichon** [comment]
@chiukapoor you should now be able to rebase as both #2773 and #2894 have merged. Note that #2894 added a new use of ConvertResourceGroupNameToLower that needs to be replaced.

File: util/azure/azure.go [hunk 1]
```diff
@@ -17,28 +17,12 @@ limitations under the License.
...
 )
 
-var azureResourceGroupNameRE = regexp.MustCompile(`.*/subscriptions/(?:.*)/resourceGroups/(.+)/providers/(?:.*)`)
-
 // AzureSystemNodeLabelPrefix is a standard node label prefix for Azure features, e.g., kubernetes.azure.com/scalesetpriority.
 const AzureSystemNodeLabelPrefix = "kubernetes.azure.com"
 
-// ConvertResourceGroupNameToLower converts the resource group name in the resource ID to be lowered.
-// Inspired by https://github.com/kubernetes-sigs/cloud-provider-azure/blob/88c9b89611e7c1fcbd39266928cce8406eb0e728/pkg/provider/azure_wrap.go#L409
-func ConvertResourceGroupNameToLower(resourceID string) (string, error) {
```
Comment (nawazkh) [inline util/azure/azure.go:32]: I wonder if we want to add `// Deprecated :` marker here and then remove this function in next minor release

File: go.mod [hunk 2]
```diff
@@ -35,15 +35,16 @@ require (
 	go.opentelemetry.io/otel/trace v1.4.0
 	golang.org/x/crypto v0.1.0
 	golang.org/x/mod v0.6.0-dev.0.20220419223038-86c51ed26bb4
-	golang.org/x/text v0.4.0
+	golang.org/x/text v0.5.0
```
Comment (nawazkh) [inline go.mod:38]: I am not so sure if minor version upgrade for `golang.org/x/text` is needed here?

**Review by nawazkh — COMMENTED** [review]
Just a thought and a question, everything else looks good to me! Thanks @chiukapoor 🚀 

- All the five local references of `ConvertResourceGroupNameToLower()` have been replaced. ✅ 
- `util/azure/azure_test.go` test file is being removed since CAPZ will not be maintaining the `ConvertResourceGroupNameToLower()` function.

File: go.mod (hunk 2 above)
Comment (CecileRobertMichon) [reply go.mod:38]: I'm guessing it came with the cloud-provider dependency, @chiukapoor to confirm

File: util/azure/azure.go (hunk 1 above)
Comment (CecileRobertMichon) [reply util/azure/azure.go:32]: My vote: let's just remove it. We don't make any guarantees on using CAPZ as a library and if anyone else happened to be using this function they can just use the cloud-provider-azure one instead.

File: go.mod (hunk 2 above)
Comment (chiukapoor) [reply go.mod:38]: Yes, it came with the `cloud-provider-azure` dependency

https://github.com/kubernetes-sigs/cloud-provider-azure/blob/bf1d7f3b46068081d795f30d440ef0997a3e9209/go.mod#L23

**Comment by nawazkh** [comment]
Nice work @chiukapoor ! 
/lgtm 🚢

**Comment by nawazkh** [comment]
^k8s-ci-bot didn't like my ship; so
🚢 
/lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**